openshift/fluentd.dc.yaml

```diff
@@ -0,0 +1,308 @@
+          # </transport>
+
+        </source>
+
+        # Recieve events from 24224/tcp
+        # this is the preferred method for sending logs between fluent-bit and fluent-d peers
+        # not currently working accross multiple namespaces
+        #<source>
+        #  @type forward
+        #  port 24224
+        #  bind 0.0.0.0
+        #</source>
+
+
+        ### Process Error logs
+        # filter logs from CDOGS (tagged with 'cdogs_node') to include only errors
+        # and rewrite tag to cdogs_node.error
+        <match cdogs_node>
+          @type rewrite_tag_filter
+          capitalize_regex_backreference yes
+          <rule>
+            key     code
+            pattern ^4\d\d$
+            tag     cdogs_node.error
+          </rule>
+        </match>
+
+        # # alternative method to filter CDOGS logs for error notifications
+        # <filter cdogs_node>
+        #   @type grep
+        #   # include only code = 40* logs
+        #   <regexp>
+        #     key code
+        #     pattern /40/
+        #   </regexp>
+        #   # see: https://docs.fluentd.org/how-to-guides/filter-modify-apache
+        # </filter>
+
+        # reformat dogs_node.error logs for output via notification
+        <filter cdogs_node.error>
+
+          @type record_transformer
+          # add a description of the error to a new key named 'content', and a 'username' required by Discord
+          enable_ruby
+          <record>
+            content ${record["product"]} - HTTP response status code ${record["code"]}, hostname: ${hostname}, agent: ${record["agent"]}, method: ${record["method"]}, path: ${record["path"]}
+            username ${record["product"]} (test) Error Alert
+          </record>
+
+          # we can remove unwanted fields from the log this way:
+          #@type record_transformer
+          #remove_keys hostname,host,user,method,path,code,size,referer,agent
+        </filter>
+
+        ### output CDOGS errors to Discord AND stdout
+        <match cdogs_node.error>
+          # You need to use the 'copy' output plugin to output the same log to multiple sources
+          @type copy
+          <store>
+            # output using out_http plugin
+            @type http
+
+            # Discord webhook
+            endpoint ${fluentd-alert-endpoint}
+            # for testing:
+            # endpoint https://fluentd-test.free.beeceptor.com
+
+            open_timeout 2
+            json_array false
+            content_type application/json
+            headers {"Content-Type":"application/json"}
+            <format>
+              @type json
+            </format>
+            <buffer>
+              flush_interval 2s
+            </buffer>
+          </store>
+
+          # send to stdout as well
+          <store>
+            @type stdout
+          </store>
+        </match>
+
+
+        ### output all other logs to standard out
+        <match **>
+          @type stdout
+        </match>
```

Comment:
After quickly scanning through the conf contents, it's probably feasible (and would make more sense) to have this be a standalone `fluent.conf.example` file in source code. Then we can leverage the oc cli directly to generate a configmap from a file instead and maybe not even have to worry about making a template out of it.

Reference: https://docs.openshift.com/container-platform/3.11/dev_guide/configmaps.html#configmaps-creating-from-files

Reply:
good call. i'll but the examples in the wiki instead